aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,375 @@
+            {
+                entry.link_label: entry.node.get_hash()
+                for entry in
+                self.get_incoming(link_type=(LinkType.INPUT_CALC, LinkType.INPUT_WORK))
+                if entry.link_label not in self._hash_ignored_inputs
+            }
+        ]
+        return objects
+
+    monkeypatch.setattr(Code, "_get_objects_to_hash", mock_objects_to_hash_code)
+    monkeypatch.setattr(CalcJobNode, "_get_objects_to_hash", mock_objects_to_hash_calcjob)
+
+    # for all other data, since they include the version
+
+    def mock_objects_to_hash(self):
+        """
+        Return a list of objects which should be included in the hash of all Nodes.
+        """
+        ignored = list(self._hash_ignored_attributes)  # pylint: disable=protected-access
+        ignored.append('version')
+        self._hash_ignored_attributes = tuple(ignored)  # pylint: disable=protected-access
+
+        objects = [
+            #importlib.import_module(self.__module__.split('.', 1)[0]).__version__,
+            {
+                key: val
+                for key, val in self.attributes_items() if key not in self._hash_ignored_attributes
+                and key not in self._updatable_attributes
+            },
+            #self._repository._get_base_folder(),
+            #self.computer.uuid if self.computer is not None else None
+        ]
+
+        return objects
+
+    # since we still want versioning for plugin datatypes and calcs we only monkeypatch aiida datatypes
+    classes_to_patch = [Dict, SinglefileData, List, FolderData, RemoteData]
+    for classe in classes_to_patch:
+        monkeypatch.setattr(classe, "_get_objects_to_hash", mock_objects_to_hash)
+
+    #BaseData, List, Array, ...
+
+
+@pytest.fixture(scope='function')
+def run_with_cache(export_cache, load_cache):
+    """
+    Fixture to automatically import an aiida graph for a given process builder.
+    """
+    def _run_with_cache(
+        builder: dict,  #aiida process builder class, or dict, if process class is given
+        process_class=None,
+        label: str = '',
+        data_dir: ty.Union[str, pathlib.Path] = 'data_dir',
+    ):
+        """
+        Function, which checks if a aiida export for a given Process builder exists,
+        if it does it imports the aiida graph and runs the builder with caching.
+        If the cache does not exists, it still runs the builder but creates an
+        export afterwards.
+
+        Inputs:
+
+        builder : AiiDA Process builder class,
+        data_dir: optional
+            Absolute path of the directory where the exported workchain graphs are
+            stored.
+
+        ignore_nodes : list string, ignore input nodes with these labels/link labels to ignore in hash.
+        # needed?
+        """
+
+        cache_exists = False
+        if data_dir == 'data_dir':
+            cwd = pathlib.Path(os.getcwd())  # Might be not the best idea.
```

Comment:
Yeah, using `os.getwcd()` is definitely not great -- the issue is that one might run `pytest` from different directories (e.g., form the top-level module directory or inside the `tests` directory). 

In my opinion, both should always work. There are two approaches to this that I've seen:
- In the (top-level) `conftest.py`, make use of `__file__` to define a fixture that gives you a constant "anchor directory". See e.g. https://github.com/FrescolinoGroup/pyhdf5io/blob/master/tests/conftest.py#L17
- Use the location of the test that you are currently running, and define paths relative to that. This is what `pytest-datadir` uses.

Both are valid, depending on the use case (do you want to share data between tests, or have it only for a specific test / test directory. I think we should leave this choice up to the user.

Reply:
I will change this. I was not familiar with pathlib and why to use it at all and not os. Same goes for the tests

Reply:
we use the package file path for our plugins. I also did not like cwd. Since relative paths are to cwd, the default is ensured to work, which was the only point in doing so.